Tests/Integration/DevAddrCacheTest.cs

```diff
@@ -335,11 +336,12 @@ public async Task Multi_Gateway_When_DevAddr_Is_In_Cache_Without_Key_Should_Not_
             await LockDevAddrHelper.PrepareLocksForTests(this.cache, lockToTake);
 
             var deviceGetter = new DeviceGetter(registryManagerMock.Object, this.cache);
+            var devNonce = new DevNonce(0xABCD);
             var tasks = new Task[3]
             {
-                deviceGetter.GetDeviceList(null, gatewayId, "ABCD", devAddrJoining),
-                deviceGetter.GetDeviceList(null, gatewayId, "ABCD", devAddrJoining),
-                deviceGetter.GetDeviceList(null, gatewayId, "ABCD", devAddrJoining),
+                deviceGetter.GetDeviceList(null, gatewayId, devNonce, devAddrJoining),
```

Comment:
Like the previous case, we can simplify this via a query so the dev nonce is repeated once:

```c#
var tasks =
    from gw in Enumerable.Repeat(gatewayId, 3)
    select deviceGetter.GetDeviceList(null, gw, new DevNonce(0xABCD), devAddrJoining);
```

Reply:
794f3b35b48935865ef9badb14c0adff9d05e6b2